app/src/main/java/ch/sdp/vibester/fragment/GameSetupFragment.kt

```diff
@@ -220,25 +226,34 @@ class GameSetupFragment : Fragment(R.layout.fragment_layout_game_setup), Adapter
      * @param tag: tag (genre) to fetch songs from: used in BY_TAG method
      * @param mode: official game mode name
      */
-    private fun chooseGenre(method: String = "", artist: String = "", tag: String = "", mode: Int = 0) {
+    private fun chooseGenre(method: String = "", artist: String = "", tag: String = "", mode: Int = 0, playOffline: Boolean = false) {
         if (artist != "") {
             val uri = LastfmUri()
 
             uri.method = method
             uri.artist = artist
             uri.tag = tag
 
-            toggleViewsVisibility(
-                goneView = vmGameSetup.view.findViewById<LinearLayout>(R.id.genrePerScoreboard),
-                visibleView = vmGameSetup.view.findViewById<ConstraintLayout>(R.id.chooseSetting)
-            )
+            if (playOffline) {
+                toggleViewsVisibility(
+                    goneView = requireView().findViewById<LinearLayout>(R.id.chooseGame),
+                    visibleView = requireView().findViewById<ConstraintLayout>(R.id.chooseSetting)
+                )
+            } else{
+                toggleViewsVisibility(
+                    goneView = requireView().findViewById<LinearLayout>(R.id.genrePerScoreboard),
+                    visibleView = requireView().findViewById<ConstraintLayout>(R.id.chooseSetting)
```

Comment:
For the CodeClimate issue, we can make the toggleViewsVisibility function take just a single boolean if the two arguments are always the same but switched around(i.e instead of tVV(a, b) you call tVV(b, a)) so the duplication can be fixed. Not an urgent thing, but could be useful if we want to refactor in the future